Automattic/wp-calypso
# Campaign details charts for Spend, Impressions and Clicks

## Proposed Changes

This adds the new charts found in the PRD for improving campaign metrics (pedfti-7HS-p2)

This is the first of several PRs, treat this PR like a checkpoint and unless there changes that should block us for shipping, I'd like to follow up with suggestions in subsequent PRs so we can ship this improvement sooner.

Belo are the supplied designs
![386186896-02aa0fa6-4529-4404-8f9a-f43f1305a9a0](https://github.com/user-attachments/assets/b2b4f8ea-31cf-4c0a-a56e-d72ca0f81523)


There are some small deviations from the design for speed (highlighted red), we can iterate on that in subsequent PRs
- The tooltip is not implemented, I'm just using the chart default (see the chart in the video demo below)
- I'm not showing stats for the campaign duration (see how they grey out in the designs) I don't think this is practical for things like evergreen campaigns.  From testing I find the charts also look nicer with shorter date ranges, so I think we should skip this part
- There is no date filtering yet, it defaults to whole campaign


## Testing Instructions

- Open up several completed campaigns, you can use the calypso live link int he comments.  I suggest a variety - short campaigns, long campaigns cancelled campaigns etc.
- I've intentionally skipped adding a loading indicator. Since Jetpack sites can't access the new endpoint yet. So I just hide the data completely unless it exists for now
- Hover to show breakdowns for each day
- Toggle between clicks and impressions using the button above that graph
- Change the colour scheme for your user - the graph colours should update 

Note: I made some subtle styling tweaks since I recorded this, but is 99% the same

https://github.com/user-attachments/assets/fd748798-ab61-4ece-a16a-240cd01b3ee5


I have also displayed evergreen data in the following way (design only covered fixed length)

**Active**
![Screenshot 2024-11-15 at 12 00 33](https://github.com/user-attachments/assets/f20e0008-8565-4da5-89d2-70ce49dd77ce)

**Completed**
![Screenshot 2024-11-15 at 12 01 02](https://github.com/user-attachments/assets/b5d00ee1-6991-46ba-a664-bbd97a3de565)


## Pre-merge Checklist
